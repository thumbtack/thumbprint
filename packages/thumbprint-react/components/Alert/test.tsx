import React from 'react';
import { mount } from 'enzyme';
import { BannerAlert, InPageAlert } from './index';

describe('BannerAlert', () => {
    test('renders children', () => {
        const wrapper = mount(
            <BannerAlert>
                <p data-test-id="text">Alert content</p>
            </BannerAlert>,
        );
        expect(wrapper.find('[data-test-id="text"]').text()).toBe('Alert content');
        expect(wrapper).toMatchSnapshot();
    });
});

describe('InPageAlert', () => {
    test('renders icon', () => {
        const wrapper = mount(
            <InPageAlert>
                <p>Alert content</p>
            </InPageAlert>,
        );
        expect(wrapper.find('svg').hasClass('icon')).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders children', () => {
        const wrapper = mount(
            <InPageAlert>
                <p data-test-id="text">Alert content</p>
            </InPageAlert>,
        );
        expect(wrapper.find('[data-test-id="text"]').text()).toBe('Alert content');
        expect(wrapper).toMatchSnapshot();
    });
});
