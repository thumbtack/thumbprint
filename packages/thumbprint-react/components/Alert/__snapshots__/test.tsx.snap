// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerAlert renders children 1`] = `
<BannerAlert>
  <div
    className="root banner info"
  >
    <div
      className="text"
    >
      <p
        data-test-id="text"
      >
        Alert content
      </p>
    </div>
  </div>
</BannerAlert>
`;

exports[`InPageAlert renders children 1`] = `
<InPageAlert>
  <div
    className="root info"
  >
    <AlertInfo
      className="icon"
      fill="currentColor"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.9994997,7.99718612 C13.4462116,7.99718612 12.9989786,8.44540556 12.9989786,8.99767595 C12.9989786,9.54994633 13.4462116,9.99816577 13.9994997,9.99816577 C14.5537884,9.99816577 15.0000208,9.54994633 15.0000208,8.99767595 C15.0000208,8.44540556 14.5537884,7.99718612 13.9994997,7.99718612 M21.0741845,21.0745886 C17.1751537,24.9754984 10.8248463,24.9754984 6.92581553,21.0745886 C3.0227827,17.1736788 3.0227827,10.8265713 6.92581553,6.92566152 C10.8258468,3.02575219 17.1741532,3.0247517 21.0741845,6.92566152 C24.9772173,10.8265713 24.9772173,17.1736788 21.0741845,21.0745886 M22.4889213,5.51096891 C17.8084836,0.829677029 10.1915164,0.829677029 5.51107869,5.51096891 C0.829640438,10.1912603 0.829640438,17.8089898 5.51107869,22.4892812 C7.85029703,24.8304274 10.9258989,26 13.9994997,26 C17.0741011,26 20.1487024,24.8304274 22.4889213,22.4892812 C27.1703596,17.8089898 27.1703596,10.1912603 22.4889213,5.51096891 M13.9994997,11.9991454 C13.4462116,11.9991454 12.9989786,12.4473649 12.9989786,12.9996352 L12.9989786,19.0025742 C12.9989786,19.5548446 13.4462116,20.003064 13.9994997,20.003064 C14.5537884,20.003064 15.0000208,19.5548446 15.0000208,19.0025742 L15.0000208,12.9996352 C15.0000208,12.4473649 14.5537884,11.9991454 13.9994997,11.9991454"
          fillRule="evenodd"
        />
      </svg>
    </AlertInfo>
    <div
      className="text"
    >
      <p
        data-test-id="text"
      >
        Alert content
      </p>
    </div>
  </div>
</InPageAlert>
`;

exports[`InPageAlert renders icon 1`] = `
<InPageAlert>
  <div
    className="root info"
  >
    <AlertInfo
      className="icon"
      fill="currentColor"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <svg
        className="icon"
        fill="currentColor"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.9994997,7.99718612 C13.4462116,7.99718612 12.9989786,8.44540556 12.9989786,8.99767595 C12.9989786,9.54994633 13.4462116,9.99816577 13.9994997,9.99816577 C14.5537884,9.99816577 15.0000208,9.54994633 15.0000208,8.99767595 C15.0000208,8.44540556 14.5537884,7.99718612 13.9994997,7.99718612 M21.0741845,21.0745886 C17.1751537,24.9754984 10.8248463,24.9754984 6.92581553,21.0745886 C3.0227827,17.1736788 3.0227827,10.8265713 6.92581553,6.92566152 C10.8258468,3.02575219 17.1741532,3.0247517 21.0741845,6.92566152 C24.9772173,10.8265713 24.9772173,17.1736788 21.0741845,21.0745886 M22.4889213,5.51096891 C17.8084836,0.829677029 10.1915164,0.829677029 5.51107869,5.51096891 C0.829640438,10.1912603 0.829640438,17.8089898 5.51107869,22.4892812 C7.85029703,24.8304274 10.9258989,26 13.9994997,26 C17.0741011,26 20.1487024,24.8304274 22.4889213,22.4892812 C27.1703596,17.8089898 27.1703596,10.1912603 22.4889213,5.51096891 M13.9994997,11.9991454 C13.4462116,11.9991454 12.9989786,12.4473649 12.9989786,12.9996352 L12.9989786,19.0025742 C12.9989786,19.5548446 13.4462116,20.003064 13.9994997,20.003064 C14.5537884,20.003064 15.0000208,19.5548446 15.0000208,19.0025742 L15.0000208,12.9996352 C15.0000208,12.4473649 14.5537884,11.9991454 13.9994997,11.9991454"
          fillRule="evenodd"
        />
      </svg>
    </AlertInfo>
    <div
      className="text"
    >
      <p>
        Alert content
      </p>
    </div>
  </div>
</InPageAlert>
`;
